mixpanel/__init__.py

```diff
@@ -429,17 +437,23 @@ def flush(self):
         :raises: MixpanelException
         """
         for endpoint in self._buffers.keys():
-            self._flush_endpoint(endpoint)
+            if endpoint == 'imports':
+                for api_key in self._buffers['imports'].keys():
+                    self._flush(endpoint, self._buffers['imports'][api_key], api_key)
+            else:
+                self._flush(endpoint, self._buffers[endpoint])
 
-    def _flush_endpoint(self, endpoint):
-        buf = self._buffers[endpoint]
+    def _flush(self, endpoint, buf, api_key=None):
         while buf:
             batch = buf[:self._max_size]
             batch_json = '[{0}]'.format(','.join(batch))
             try:
-                self._consumer.send(endpoint, batch_json)
+                self._consumer.send(endpoint, batch_json, api_key)
```

Comment:
I need to verify that all of our built-in consumers know how to deal with a `None` endpoint, and we should probably only provide the api_key argument if we know it's not `None`- something like
```
if api_key:
    self._consumer.send(endpoint, batch_json, api_key)
else:
    self._consumer.send(endpoint, batch_json)
```

 It's a bit annoying, I know, but we advise people to build their own consumers, and those consumers might break if we call them with three arguments. (It's also probably worth a comment since it'll be weirder than what you wrote)